frontend/src/hooks/useChat.ts

```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import { BACKEND_URL } from '../config';
+import useLocalStorage from './useLocalStorage';
+
+interface ChatMessage {
+  role: 'user' | 'assistant';
+  content: string;
+}
+
+const getChatHistoryKey = (userId: string, blogId: string) => `chatHistory_${userId}_${blogId}`;
+const CHAT_LIMIT = 20;
```

Comment:
This is defined at 2 places. Instead declare it once in config.ts and reuse it in both places